src/app/loadout-builder/loadout-builder-reducer.ts

```diff
@@ -363,16 +363,10 @@ function lbConfigReducer(defs: D2ManifestDefinitions) {
         };
       }
       case 'statConstraintChanged': {
-        const { constraint: newConstraint } = action;
-        const newStatConstraints = state.resolvedStatConstraints.map((c) => {
-          if (c.statHash === newConstraint.statHash) {
-            // Previously ignored stats' maxes need to be bumped up to T10 again
-            const maxTier = newConstraint.ignored ? 0 : c.ignored ? 10 : c.maxTier;
-            return { ...newConstraint, maxTier };
-          } else {
-            return c;
-          }
-        });
+        const { constraint } = action;
+        const newStatConstraints = state.resolvedStatConstraints.map((c) =>
+          c.statHash === constraint.statHash ? constraint : c,
+        );
```

Comment:
I ended up reverting this - otherwise, you lose information about what was set when you toggle. I'm not sure if that breaks anything about the original PR - if it does, we could add a bit of code that transforms the max tiers on the way to the `process` handler?

Reply:
Yeah that makes sense - but it needs to be transformed on the way to the rest of the LB UI too (sets sorting, set stats) since they now only use the `maxTier` for calculations. `ignored` is basically redundant for calculation purposes (since `ignored <=> (maxTier == 0)`) and shouldn't affect correctness, it's basically only used as a performance optimization in the process.

Reply:
OK, take a look at the latest commit. I've introduced and used a new type that doesn't have `ignored`, and instead only relies on a `maxTier` value.